frontend/src/scenes/billing/BillingHero.tsx

```diff
@@ -1,26 +1,244 @@
 
-    return (
-        <div className="BillingHero" ref={billingHeroRef}>
-            <div className="p-4">
-                <p className="text-xs uppercase my-0">How pricing works</p>
-                <h1 className="ingestion-title">Get the whole hog.</h1>
-                <h1 className="ingestion-title text-danger">Only pay for what you use.</h1>
-                <p className="mt-2 mb-0">
-                    Upgrade to get access to premium product and platform features. Set billing limits as low as $0 to
-                    control spend.
+const PLAN_BADGES: Record<BillingPlan, string> = {
+    [BillingPlan.Free]: planFree,
+    [BillingPlan.Paid]: planPaid,
+    // TODO: Add teams badge once ready
+    [BillingPlan.Teams]: planPaid,
+    [BillingPlan.Enterprise]: planEnterprise,
+}
+
+const STARTUP_PROGRAM_BADGES: Record<StartupProgramLabel, string> = {
+    [StartupProgramLabel.YC]: planYc,
+    [StartupProgramLabel.Startup]: planStartup,
+}
+
+interface CopyVariation {
+    title: string | null
+    subtitle: string | null
+    getDescription: (billingPlan: BillingPlan, scrollToProduct: (productType: string) => void) => JSX.Element
+    backgroundColor: string
+}
+
+const BADGE_CONFIG: Record<BillingPlan | StartupProgramLabel, CopyVariation> = {
+    [BillingPlan.Free]: {
+        title: 'Get the whole hog.',
+        subtitle: 'Only pay for what you use.',
+        backgroundColor: 'bg-danger-highlight',
+        getDescription: () => (
+            <>
+                <p>PostHog comes with all product features on every plan.</p>
+                <p>Add your credit card to remove usage limits and unlock all platform features.</p>
+                <p>Set billing limits as low as $0 to control your spend.</p>
+                <p className="italic">P.S. You still keep the monthly free allotment for every product!</p>
+            </>
+        ),
+    },
+    [BillingPlan.Paid]: {
+        title: 'Good call!',
+        subtitle: "You're on the Ridiculously Cheap™ plan.",
+        backgroundColor: 'bg-warning-highlight',
+        getDescription: (_billingPlan: BillingPlan, scrollToProduct: (productType: string) => void) => (
+            <>
+                <p>PostHog comes with all product features on every plan.</p>
+                <p>
+                    If you're growing like crazy, you might want to check out the{' '}
+                    {scrollToProduct ? (
+                        <>
+                            <Link onClick={() => scrollToProduct('teams')}>Teams</Link>
+                            {' or '}
+                            <Link onClick={() => scrollToProduct('enterprise')}>Enterprise</Link>
+                        </>
+                    ) : (
+                        'Teams or Enterprise'
+                    )}{' '}
+                    plan.
+                </p>
+            </>
+        ),
+    },
+    [BillingPlan.Teams]: {
+        title: 'Good call!',
+        subtitle: "You're on the Teams plan.",
```

Comment:
I struggle a bit with this and I know it's complicated but technically this is an add-on.

It's tricky since technical we have free, paid, enterprise and then teams, enterprise add-ons. 

We should problem just refer to them as plans but right now we don't.

I think it's ok for now but we should bring it up in our channel and chat about it more.

Reply:
Yup agreed it's confusing, probably even more so to us than to users - they probably don't even notice those are addons and just assume they are plans.

Reply:
Ha, yeah, that's what I was thinking. It might be worth us just calling them plans in the UI.